src/Sentry/Internal/PiiExtensions.cs

```diff
@@ -0,0 +1,55 @@
+namespace Sentry.Internal;
+
+/// <summary>
+/// Extensions to help redact data that might contain Personally Identifiable Information (PII) before sending it to
+/// Sentry.
+/// </summary>
+internal static class PiiExtensions
+{
+    internal const string RedactedText = "[Filtered]";
+
+    /// <summary>
+    /// Searches for URLs in text data and redacts any PII data from these, as required.
+    /// </summary>
+    /// <param name="data">The data to be searched</param>
+    /// <returns>
+    /// The data, if no PII data is present or a copy of the data with PII data redacted otherwise
+    /// </returns>
+    public static string? RedactUrl(this string? data)
```

Comment:
Generally, prefer to let the _caller_ deal with nullability on extension methods.

```suggestion
    public static string RedactUrl(this string data)
```

So usage would be like `foo = foo?.RedactUrl()` when `foo` is a `string?`, or `foo = foo.RedactUrl()` if it's just a `string`.

Reply:
... then update usages.